src/Ratchet/Server/IoServer.php

```diff
@@ -80,61 +114,61 @@ public function run() {
      * @param \React\Socket\ConnectionInterface $conn
      */
     public function handleConnect($conn) {
-        $conn->decor = new IoConnection($conn);
-        $conn->decor->resourceId = (int)$conn->stream;
+        $io_conn = new IoConnection($conn);
+        $io_conn->resourceId = (int)$conn->stream;
 
         $uri = $conn->getRemoteAddress();
-        $conn->decor->remoteAddress = trim(
+        $io_conn->remoteAddress = trim(
             parse_url((strpos($uri, '://') === false ? 'tcp://' : '') . $uri, PHP_URL_HOST),
             '[]'
         );
 
-        $this->app->onOpen($conn->decor);
+        $this->app->onOpen($io_conn);
 
-        $conn->on('data', function ($data) use ($conn) {
-            $this->handleData($data, $conn);
+        $conn->on('data', function ($data) use ($io_conn) {
+            $this->handleData($data, $io_conn);
         });
-        $conn->on('close', function () use ($conn) {
-            $this->handleEnd($conn);
+        $conn->on('close', function () use ($io_conn) {
+            $this->handleEnd($io_conn);
         });
-        $conn->on('error', function (\Exception $e) use ($conn) {
-            $this->handleError($e, $conn);
+        $conn->on('error', function (\Exception $e) use ($io_conn) {
+            $this->handleError($e, $io_conn);
         });
     }
 
     /**
      * Data has been received from React
      * @param string                            $data
-     * @param \React\Socket\ConnectionInterface $conn
+     * @param \Ratchet\ConnectionInterface $io_conn
      */
-    public function handleData($data, $conn) {
+    public function handleData($data, $io_conn) {
```

Comment:
Changing the parameter there is a breaking change, because IoServer can be extended

Reply:
I agree, this a breaking change, however it was done because \React\Socket\ConnectionInterface doesn't actually have the decor property previously being used to pass the connection around and PHP 8 deprecated dymanic property creation.